src/main/java/fr/openent/viescolaire/service/impl/DefaultGroupingService.java

```diff
@@ -0,0 +1,160 @@
+                PromiseHelper.reject(log, messageToFormat, this.getClass().getSimpleName(), new Exception(res.left().getValue()), promise);
+
+            }
+        }));
+    return promise.future();
+    }
+
+    /**
+     * Update the grouping
+     * @param groupingId    Identifier of the grouping
+     * @param name          New grouping name
+     * @return              Promise with the status of the update
+     */
+    @Override
+    public Future<JsonObject> updateGrouping(String groupingId ,String name) {
+        Promise<JsonObject> promise = Promise.promise();
+        if (name == null || groupingId == null || groupingId.isEmpty() || name.isEmpty()) {
+            String messageToFormat = "[vie-scolaire@%s::updateGrouping] Error while updating grouping : %s";
+            PromiseHelper.reject(log, messageToFormat, this.getClass().getSimpleName(), new Exception("error.parameters"), promise);
+            return promise.future();
+        }
+        JsonArray values = new JsonArray();
+        StringBuilder query = new StringBuilder();
+        query.append("UPDATE viesco.grouping ")
+                .append("SET name = ?, updated_at = ? WHERE id = ?");
+        values.add(name);
+        values.add(DateHelper.getCurrentDate(DateHelper.MONGO_FORMAT));
+        values.add(groupingId);
+        Sql.getInstance().prepared(query.toString(), values, SqlResult.validUniqueResultHandler(res -> {
+            if (res.isRight())
+                promise.complete(new JsonObject().put(Field.STATUS, Field.OK));
+            else {
+                String messageToFormat = "[vie-scolaire@%s::updateGrouping] Error while updating grouping : %s";
+                PromiseHelper.reject(log, messageToFormat, this.getClass().getSimpleName(), new Exception(res.left().getValue()), promise);
+            }
+        }));
+        return promise.future();
+    }
+
+
+    /**
+     * Add classes and groups to the grouping
+     * @param groupingId    Identifier of the grouping
+     * @param groupId       Identifier of the group
+     * @param classId       Identifier of the class
+     * @return              Promise with the status of the operation.
+     */
+    @Override
+    public Future<JsonObject> addToGrouping(String groupingId, String groupId, String classId) {
+        Promise<JsonObject> promise = Promise.promise();
+        if (classId == null || groupId == null || groupingId == null ||classId.isEmpty() || groupingId.isEmpty() || groupId.isEmpty()) {
+            String messageToFormat = "[vie-scolaire@%s::addToGrouping] Error while adding to grouping : %s";
+            PromiseHelper.reject(log, messageToFormat, this.getClass().getSimpleName(), new Exception("error.parameters"), promise);
+            return promise.future();
+        }
+        JsonArray values = new JsonArray();
+        StringBuilder query = new StringBuilder();
+        groupAndClassExist(classId, groupId)
+                .onSuccess(res -> {
+                    if (res) {
```

Comment:
```suggestion
                   Boolean.TRUE.equals(res)
```
https://rules.sonarsource.com/java/RSPEC-5411